src/coaster.cpp

```diff
@@ -250,15 +250,15 @@ const ImageData *DisplayCoasterCar::GetSprite(const SpriteStorage *sprites, View
 	return this->car_type->GetCar(this->pitch, this->roll, (this->yaw + orient * 4) & 0xF);
 }
 
-std::vector<std::pair<const ImageData*, const Recolouring*>> DisplayCoasterCar::GetOverlays(const SpriteStorage *sprites, ViewOrientation orient) const
+VoxelObject::Overlays DisplayCoasterCar::GetOverlays(const SpriteStorage *sprites, ViewOrientation orient) const
 {
-	std::vector<std::pair<const ImageData*, const Recolouring*>> result;
+	Overlays result;
 	if (this->owning_car == nullptr) return result;
 	for (int i = 0; i < this->car_type->num_passengers; i++) {
 		if (this->owning_car->guests[i] != nullptr) {
-			result.push_back(std::make_pair(
+			result.push_back(Overlay {
```

Comment:
In C++11 iirc, there is `emplace_back` which takes the parameters of the constructor of the vector element, and does what you do here (construct an object from the parameters, and add it to the vector), except the new element is not copied.
I don't think we actually use that in the code yet, but it could be worth considering at some point.